drivers/i2c/i2c_rcar.c

```diff
@@ -307,25 +348,37 @@ static const struct i2c_driver_api i2c_rcar_driver_api = {
 };
 
 /* Device Instantiation */
-#define I2C_RCAR_INIT(n)						      \
-	static const struct i2c_rcar_cfg i2c_rcar_cfg_##n = {		      \
-		.reg_addr = DT_INST_REG_ADDR(n),			      \
-		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),           \
-		.bitrate = DT_INST_PROP(n, clock_frequency),                  \
-		.mod_clk.module =					      \
-			DT_INST_CLOCKS_CELL_BY_IDX(n, 0, module),	      \
-		.mod_clk.domain =					      \
-			DT_INST_CLOCKS_CELL_BY_IDX(n, 0, domain),	      \
-	};								      \
-									      \
-								              \
-	DEVICE_DT_INST_DEFINE(n,					      \
-			    i2c_rcar_init,				      \
-			    device_pm_control_nop,			      \
-			    NULL,			              \
-			    &i2c_rcar_cfg_##n,				      \
-			    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,  \
-			    &i2c_rcar_driver_api			      \
-			    );						      \
+#define I2C_RCAR_INIT(n)						       \
+	static void i2c_rcar_##n##_init(const struct device *dev);	       \
+	static const struct i2c_rcar_cfg i2c_rcar_cfg_##n = {		       \
+		.reg_addr = DT_INST_REG_ADDR(n),			       \
+		.init_func = i2c_rcar_##n##_init,			       \
+		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),	       \
+		.bitrate = DT_INST_PROP(n, clock_frequency),		       \
+		.mod_clk.module =					       \
+			DT_INST_CLOCKS_CELL_BY_IDX(n, 0, module),	       \
+		.mod_clk.domain =					       \
+			DT_INST_CLOCKS_CELL_BY_IDX(n, 0, domain),	       \
+	};								       \
+									       \
+	static struct i2c_rcar_data i2c_rcar_data_##n;			       \
+									       \
+	DEVICE_DT_INST_DEFINE(n,					       \
+			      i2c_rcar_init,				       \
+			      device_pm_control_nop,			       \
```

Comment:
```suggestion
			      NULL,			       \
```
https://github.com/zephyrproject-rtos/zephyr/pull/34641#event-4657433284